app/src/main/kotlin/com/wire/android/ui/authentication/login/Login.kt

```diff
@@ -58,21 +56,18 @@ import kotlinx.coroutines.launch
 
 @ExperimentalMaterialApi
 @Composable
-fun LoginScreen(
-    navController: NavController,
-    serverConfig: ServerConfig
-) {
+fun LoginScreen(serverConfig: ServerConfig) {
     val scope = rememberCoroutineScope()
     val loginViewModel: LoginViewModel = hiltViewModel()
     val loginState: LoginState = loginViewModel.loginState
     LoginContent(
         loginState = loginState,
         onUserIdentifierChange = { loginViewModel.onUserIdentifierChange(it) },
-        onBackPressed = { navController.popBackStack() },
+        onBackPressed = { loginViewModel.navigateBack() },
         onPasswordChange = { loginViewModel.onPasswordChange(it) },
         onDialogDismiss = { loginViewModel.clearLoginError() },
         onRemoveDeviceOpen = {
-            navController.navigate(AuthDestination.removeDeviceScreen)
+            loginViewModel.onTooManyDevicesError()
             loginViewModel.clearLoginError()
```

Comment:
I think it would make more sense to call clearLoginError() inside of the "broader" method inside the viewmodel